UnitySDK/Assets/ML-Agents/Scripts/RewardProvider/IRewardProvider.cs

```diff
@@ -0,0 +1,36 @@
+namespace MLAgents.RewardProvider
+{
+    /// <summary>
+    /// Reward providers allow users to provide rewards for Agent behavior during training in order to
+    /// give hints on what types of actions are "better" than others based on an Agent's previous observation.
+    /// </summary>
+    public interface IRewardProvider
+    {
+        /// <summary>
+        /// Get an incremental reward to pass along to a trainer.  
+        /// </summary>
+        /// <returns></returns>
+        float GetIncrementalReward();
+
+        /// <summary>
+        /// Adds a scalar value to the current reward for this step.
+        /// </summary>
+        void AddReward(float reward);
+
+        void SetReward(float reward);
+
+        /// <summary>
+        /// Retrieves the step reward for the Agent.
+        /// </summary>
+        /// <returns>The step reward.</returns>
+        float GetReward();
+        
+        /// <summary>
+        /// Retrieves the episode reward for the Agent.
+        /// </summary>
+        /// <returns>The episode reward.</returns>
+        float GetCumulativeReward();
```

Comment:
I am rather a fan of small interfaces and I am wondering if GetCumulativeReward should be on this interface. In a lot of application, we do not want to know about the cumulative reward (except for debugging and for our internal use) so I am wondering if we should remove them from the interface and wrap the reward provider of the Agent into a Reward provider that gives access to the cumulative reward (and additionally can log it).

Reply:
could be removed from the interface, and left in the implementation.